### DISCUSSION TITLE
Why the TensorRT mdoel doesn't show class names(model.names)?

### DISCUSSION THREAD

**Discussion by miladfa7** [ultralytics/ultralytics#4461] [Q&A]
I have converted the model to TensorRT format using this code

`model.export(format='engine',  half=True, imgsz=[544,960], batch=10, device=0)`

but after exporting model, I want to see class names but it shows just `None` value as result. 

```
tensorrt_path ="train4/weights/best.engine"
trt_model = YOLO(tensorrt_path)
print(trt_model.names)
```  
**Output**: 
`None`

Also when the tensorrt model used for inference, the model has not detected any object. `results` model was empty.  
I mean. 
`results = trt_model.predict(frames, imgsz=[544,960], batch=10, device=0)`

**Comment by glenn-jocher** [comment]
@miladfa7 hello! It seems you're encountering an issue where the class names are not being displayed after converting a YOLOv8 model to TensorRT format, and you're also not getting any detections.

For the class names issue, after exporting the model to TensorRT, the class names are not automatically associated with the `.engine` file. You'll need to manually load the class names from the dataset's `.yaml` file or another source where they are defined.

Regarding the lack of detections, there could be several reasons for this. It's important to ensure that the model was exported correctly and that the input data is preprocessed in the same way as during training. Additionally, verify that the confidence threshold and non-maximum suppression (NMS) settings are appropriate for your use case.

If you continue to face issues, please refer to our documentation on the Predict mode and the Troubleshooting section for guidance on common problems and solutions. If the problem persists, feel free to open an issue on the repo with detailed information about your setup and steps to reproduce the issue, and we'll be happy to help you further. 😊
